### ISSUE TITLE
ipn_main_handler:  Fatal error on chargeback reimbursement (missing mc_fee)

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#4493]
As identified in [this](https://www.zen-cart.com/showthread.php?228502-chargeback_reimbursement-error-log) Zen Cart forum posting, PayPal chargebacks don't include an `mc_fee` element.  That then results in a PHP Fatal error:
```
[28-Nov-2021 09:47:05 America/New_York] Request URI: /ipn_main_handler.php, IP address: 173.0.81.65
#1 trigger_error() called at [/includes/classes/db/mysql/query_factory.php:170]
#2 queryFactory->show_error() called at [/includes/classes/db/mysql/query_factory.php:142]
#3 queryFactory->set_error() called at [/includes/classes/db/mysql/query_factory.php:269]
#4 queryFactory->Execute() called at [/includes/functions/database.php:44]
#5 zen_db_perform() called at [/ipn_main_handler.php:419]
--> PHP Fatal error: 1366:Incorrect decimal value: '' for column 'mc_fee' at row 1 :: INSERT INTO paypal (order_id, txn_type, module_name, module_mode, reason_code, payment_type, payment_status, pending_reason, invoice, mc_currency, first_name, last_name, payer_business_name, address_name, address_street, address_city, address_state, address_zip, address_country, address_status, payer_email, payer_id, payer_status, payment_date, business, receiver_email, receiver_id, txn_id, parent_txn_id, num_cart_items, mc_gross, mc_fee, settle_amount, settle_currency, exchange_rate, notify_version, verify_sign, date_added, memo) VALUES ('20744', 'parent', 'paypal (ipn-handler)', 'IPN', 'chargeback_reimbursement', '', 'Completed', '', '3165-1628894647-[Mystore]', 'USD', '', '', '', '', '', '', '', '', '', '', 'me@myemail.com', 'Z6PFML62KA2D4', 'verified', '2021-11-26 09:09:56', '', '', '', '49L67964SW698244C', '8C060304EJ230962J', '1', '225.85', '', '0', '', '1', '3.9', 'AtUbgugC4KybUOS9Xzrs4wqyfk7uAUAqom0B9gFibfjxN5omDN9aT-O-', now(), '{Record generated by IPN}') ==> (as called by) /includes/functions/database.php on line 44 <== in /includes/classes/db/mysql/query_factory.php on line 170.
```

**Version**
Zen Cart version:  1.5.7c

**Comment by scottcwilson** [comment]
Could this be fixed by changing in 1.5.8 includes/modules/payment/paypal/paypal_functions.php line 305

                            'mc_fee' => $_POST['mc_fee'],

to

                            'mc_fee' => $_POST['mc_fee'] ?? 0,

**Comment by lat9** [comment]
> Could this be fixed by changing in 1.5.8 includes/modules/payment/paypal/paypal_functions.php line 305
> 
> ```
>                         'mc_fee' => $_POST['mc_fee'],
> ```
> 
> to
> 
> ```
>                         'mc_fee' => $_POST['mc_fee'] ?? 0, 
> ```

Go for it!
